frontend/src/js/previous-queries/list/ShareProjectItemModal.tsx

```diff
@@ -110,28 +117,32 @@ const ShareProjectItemModal = ({ item, onClose }: PropsT) => {
       ? t("sharePreviousQueryModal.unshare")
       : t("common.share");
 
+  const buttonDisabled =
+    JSON.stringify(initialUserGroupsValue) === JSON.stringify(userGroupsValue);
```

Comment:
This seems rather hacky to me, relying on the serialization of the state object? Wouldn't it be preferable to use an elementwise comparison instead (likely also much more efficient)

Reply:
I understand your point, but user groups are short arrays, this is efficient enough, so I'd rather avoid micro optimizations here

Reply:
My point isn't as much about optimization than about misappropriating serialization for something as basic as equality checks. But if you'd rather keep it I'm fine with it.

Reply:
Agreed that it feels weird until you've seen it lots of times in JS code. It's more common than you'd think: https://stackoverflow.com/a/1144249/921486